tests/base/develop_pipelines.py

```diff
@@ -47,10 +47,14 @@ def run_model_test(trainer_options, model, on_gpu: bool = True, version=None, wi
         trainer_options.update(checkpoint_callback=True)
 
     trainer = Trainer(**trainer_options)
+    initial_values = torch.tensor([torch.sum(torch.abs(x)) for x in model.parameters()])
     result = trainer.fit(model)
+    post_train_values = torch.tensor([torch.sum(torch.abs(x)) for x in model.parameters()])
 
     # correct result and ok accuracy
     assert result == 1, 'trainer failed'
+    # Check that the model is actually changed post-training
+    assert torch.norm(initial_values - post_train_values) > 0.1
```

Comment:
Since we are working with discrete values here, I conjecture the probability that the model changes and lands at the same weights at the end of training is strictly > 0 😄